lib/ansible/modules/database/mongodb/mongodb_role.py

```diff
@@ -0,0 +1,452 @@
+        from pymongo import Connection as MongoClient
+    except ImportError:
+        HAS_PYMONGO = False
+    else:
+        HAS_PYMONGO = True
+else:
+    HAS_PYMONGO = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves import configparser
+from ansible.module_utils._text import to_native
+
+
+# =========================================
+# MongoDB module specific support methods.
+#
+
+def check_compatibility(module, client):
+    """Check the compatibility between the driver and the database.
+
+       See: https://docs.mongodb.com/ecosystem/drivers/driver-compatibility-reference/#python-driver-compatibility
+
+    Args:
+        module: Ansible module.
+        client (cursor): Mongodb cursor on admin database.
+    """
+    loose_srv_version = LooseVersion(client.server_info()['version'])
+    loose_driver_version = LooseVersion(PyMongoVersion)
+
+    if loose_srv_version >= LooseVersion('3.2') and loose_driver_version < LooseVersion('3.2'):
+        module.fail_json(msg=' (Note: you must use pymongo 3.2+ with MongoDB >= 3.2)')
+
+    elif loose_srv_version >= LooseVersion('3.0') and loose_driver_version <= LooseVersion('2.8'):
+        module.fail_json(msg=' (Note: you must use pymongo 2.8+ with MongoDB 3.0)')
+
+    elif loose_srv_version >= LooseVersion('2.6') and loose_driver_version <= LooseVersion('2.7'):
+        module.fail_json(msg=' (Note: you must use pymongo 2.7+ with MongoDB 2.6)')
+
+    elif LooseVersion(PyMongoVersion) <= LooseVersion('2.5'):
+        module.fail_json(msg=' (Note: you must be on mongodb 2.4+ and pymongo 2.5+ to use the roles param)')
+
+
+def load_mongocnf(config_file):
+    config = configparser.RawConfigParser()
+
+    try:
+        with open(config_file) as f:
+            config.readfp(f)
+            creds = dict(
+                user=config.get('client', 'user'),
+                password=config.get('client', 'pass')
+            )
+    except (configparser.NoOptionError, IOError):
+        return False
+
+    return creds
+
+
+def check_authentication_restrictions_support(client):
+    loose_srv_version = LooseVersion(client.server_info()['version'])
+    return loose_srv_version >= LooseVersion('4.0')
+
+
+def role_find(module, client, db_name, role, authentication_restrictions_supported):
+    db = client[db_name]
+
+    if authentication_restrictions_supported:
+        # yes, showAuthenticationRestrictions *is* undocumented
+        result = db.command('rolesInfo',
+                            role,
+                            showPrivileges=True,
+                            showAuthenticationRestrictions=True
+                            )
+    else:
+        result = db.command('rolesInfo',
+                            role,
+                            showPrivileges=True
+                            )
+
+    if result['ok'] != 1:
+        module.fail_json(msg='rolesInfo failed', exception=traceback.format_exc())
```

Comment:
This message is not very informative. Maybe there's some data you'd like to add to the returned result?